pkg/cmd/promhelperservice/handlers/instanceconfigs/delete.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2024 The Cockroach Authors.
+
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package instanceconfigs
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/cockroachdb/cockroach/pkg/cmd/promhelperservice/logging"
+)
+
+// DeleteHandler handles DELETE /instance-configs/{cluster_id}
+// Doc reference - https://cockroachlabs.atlassian.net/wiki/x/MAAlzg
+func DeleteHandler(w http.ResponseWriter, r *http.Request) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
```

Comment:
Nit (here and in other places): if we are never canceling the context other than in the `defer` call, then the resulting context is no different than `context.Background()`. Might as well just have a `ctx := context.Background()` at the top of these functions and use it when needed.